index.js

```diff
@@ -238,22 +238,29 @@ ShareDbMongo.prototype.close = function(callback) {
 
 ShareDbMongo.prototype.commit = function(collectionName, id, op, snapshot, options, callback) {
   var self = this;
-  var request = createRequestForMiddleware(options, collectionName, op);
+  var request = createRequestForMiddleware(null, options, collectionName, op);
   this._writeOp(collectionName, id, op, snapshot, function(err, result) {
     if (err) return callback(err);
     var opId = result.insertedId;
     self._writeSnapshot(request, id, snapshot, opId, function(err, succeeded) {
       if (succeeded) return callback(err, succeeded);
       // Cleanup unsuccessful op if snapshot write failed. This is not
-      // neccessary for data correctness, but it gets rid of clutter
+      // necessary for data correctness, but it gets rid of clutter
       self._deleteOp(request.collectionName, opId, function(removeErr) {
         callback(err || removeErr, succeeded);
       });
     });
   });
 };
 
-function createRequestForMiddleware(options, collectionName, op) {
+function createRequestForMiddleware(fields, options, collectionName, op) {
```

Comment:
Let's change the order of the params to more closely match the typical order when calling the exported functions:
```suggestion
function createRequestForMiddleware(collectionName, fields, options, op) {
```

`op` at the end since some invocations don't provide it.

I didn't really notice the order inconsistency before with fewer params, but it's bugging me now 😅

Reply:
oh ok. Keep in mind function isn't exported! :)

Reply:
updated!